File: 2013/Lunch & Learn/IntroToOOP/VB.NET/IntroToOOP/Example1/Cat.vb

Namespace IntroToOOP.Example1
    Public Class Cat

        'Internal Members
        Private _name As String
        Private _color As String

        'Constructor
        Public Sub New()
            _name = "Sue"
            _color = "Black"
        End Sub

        Public Sub New(name As String, color As String)
            _name = name
            _color = color
        End Sub

        Property Name As String
            Get
                Return _name
            End Get
            Set(value As String)
                _name = value
            End Set
        End Property

        ReadOnly Property Color As String
            Get
                Return _color
            End Get
        End Property

        'Behavior
        Public Sub wagsTail()
            Console.WriteLine("{0} wags tail...", _name)
        End Sub

        Public Sub Speaks()
            Console.WriteLine("{0} purrs...", _name)
        End Sub


    End Class
End Namespace

File: 2013/Lunch & Learn/IntroToOOP/C#/Example1/Cat.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace IntroToOOP.Example1
{
    public class Cat
    {
         /*Internal Members*/
        private string _name;
        private string _color; 

        /* Constructor */
        public Cat()
        {
            /* State */
            _name = "Hip Cat";
            _color = "Orange";
        }

        /* Consctructor with parameters (different signature) */
        public Cat(string name, string color)
        {
            _name = name;
            _color = color;
        }

        /* Public properties to allow the return and modification of the private members
         * Allow the State to be changed from outside the class*/

        /* State */
        public string Name
        {
            get
            {
                return _name;
            }

            set
            {
                _name = value;
            }
        }

        /* read-only */
        public string Color
        {
            get
            {
                return _color;
            }          
        }

        /* Behavior */
        public void wagsTail()
        {
            Console.WriteLine("{0} wags tail...", _name);
        }

        public void Speaks()
        {
            Console.WriteLine("{0} purrs...", _name);
        }

    }
}
